app/lib/activitypub/activity/create.rb

```diff
@@ -95,19 +95,21 @@ def find_existing_status
   end
 
   def process_status_params
+    @status_parser = ActivityPub::Parser::StatusParser.new(@json, followers_collection: @account.followers_url)
+
     @params = begin
       {
-        uri: object_uri,
-        url: object_url || object_uri,
+        uri: @status_parser.uri,
+        url: @status_parser.url || @status_parser.uri,
```

Comment:
The only things `object_uri` does is choose between treating the `object` as the ID if it's a string or take the `id` from it if it's an object, and then parse out bearcaps. Sanity checks do not go through `object_uri`. Since by the point the parser is used, `object` is definitely not a string anymore, it's fine to use the parser. I added bearcap handling to the parser.